diff_settings.py

```diff
@@ -1,6 +1,10 @@
+def add_custom_arguments(parser):
+    parser.add_argument("-v", "--version", help="Game version to diff", default="gc-eu-mq-dbg")
```

Comment:
This is why I wanted to make this change in a separate PR 🙃 

I changed the log form to `--oot-version`, but I don't want to give up on `-v`. I strongly feel that it's important for this argument to be short and intuitive since we'll be typing it a lot, and I don't think any other letter would work. If some tool does someday get a `--verbose` option, I'm okay not having a short option for that, and I don't think people would confuse this for asm-differ's `-V`.

Reply:
I understand wanting to keep -v short but I feel it's a bit useless when you have to type the long "gc-eu-mq-dbg" or similar right after, so I don't think you would type this over and over again anyway, and would rather copypaste it

(personally I would probably end up using shell aliases or wrapper scripts)

I don't think with introducing this as `-v` we would ever move it as something else if a verbose flag was added somewhere. This is the kind of thing you cannot change, because people will be used to it.

Regardless I'm okay with it as -v as things are currently, I'm only resisting because I know it would be much harder to change later

Reply:
I think for scripts like `diff.py` or `sym_info.py`/`first_diff.py`, it would be ideal if it defaults to the last version built so you don't have to provide the version everytime. There are multiple ways to achieve this, like looking at the last modified files in build/ for instance, or tracking the version currently being worked on in a local file (this would be my preferred option since it's easy to reference anywhere)

Reply:
I also agree with the concerns about -v and --version

Reply:
The primary concern here should be usability since this is something people will interact with a lot. It's true that `-v` is often used for "software version" or "verbose" in other programs, which does make it less usable to use `-v` here for something else. But I think that is outweighed by the usability benefits of using `-v`, namely that it's nice to have a short form for such a common option, and there's no other letter that fits.

> I understand wanting to keep -v short but I feel it's a bit useless when you have to type the long "gc-eu-mq-dbg" or similar right after, so I don't think you would type this over and over again anyway, and would rather copypaste it

Well, gc-eu-mq is still much faster for me to type than copy-pasting it, and fortunately "gc-eu-mq-dbg" is the default so I never have to type it.

> I don't think with introducing this as -v we would ever move it as something else if a verbose flag was added somewhere.

Right, I meant in the off change we did add a verbose flag, I would be much happier typing out `--verbose` than giving up `-v` for the OOT version.

> I think for scripts like diff.py or sym_info.py/first_diff.py, it would be ideal if it defaults to the last version built so you don't have to provide the version everytime.

Defaulting to the last version built sounds way too magical IMO. I've been building and diffing gc-eu-mq and gc-eu-mq-dbg at the same time and having some kind of hidden state sounds very confusing. But even then I would still want a short option to switch between the two.

Reply:
(I've already changed `--version` -> `--oot-version` btw, I'm talking about the short option here)

Reply:
What about a shorter long option (alias) like `--ootv` or `--ov`

Reply:
-ov sounds like a fine compromise to me

Reply:
Eh those both feel kinda ugly to me. `--ov` could be confused for `-ov` (i.e. passing both `-o` and `-v`), and `--ootv` makes me think of https://www.twitch.tv/ootv. Not nearly as nice as `-v`, which clearly stands for "version", which in the context of this repo almost always means "OOT game version" and not "software tool version"

Reply:
-v is short for --verbose in [most] command line tools

I really do think it's worth taking into account and adhering to *de facto* standards when designing something new.
It makes less to take in for someone experienced, and is more useful / less misleading to learn for someone inexperienced.

[If I can rant for a bit this is why linux and related software is so good, developments are made to blend in instead of making flashy changes à la Microsoft because more new=more sales]

Reply:
I hear your points and I understand you feel strongly about this, but 3 people so far feel like it needs to change. We can get even more opinions on this if you want, but at the moment its pretty lopsided.

--ov as an alias, with two dashes to signify it being one argument name, still sounds like the best compromise so far

Reply:
I have taken those de facto standards into account, and decided that other factors have outweighed them. The primary goal should be to make a tool that people enjoy using, and adhering to standards is one component of that, but it's a mistake to make that the end goal.

I'm not looking for more opinions, I'm looking for a more convincing argument than "it could be confusing". Maybe this github thread is not the best place for that though, so I'm going to revert adding version flags for this PR and we can revisit it later

Reply:
Its common in this project (and I'd say group projects in general) to concede to changes you may not like for the sake of compromise with the majority opinion. Feel like it needs to be said now while you are new to contributing here, that majority opinion is our main path forward in any heated debate. 
It will not be the last time people disagree with changes you make, unfortunately, but it happens to everyone. Pushing the conversation to later wont really help I don't think.

Reply:
Don't worry, I understand that. I do think it would help though if I had a little more time to think about this and try out some options. In the meantime I don't want to hold up the rest of this PR since it's kind of blocking everything multiversion-related.

I added versions flags to this PR against my better judgement (I kinda knew this would happen) so I'm just trying to put it back. Sorry if it looks like I'm being a baby about it, I don't mean it that way

Reply:
Alright, maybe later is better in this case then. 
And no worries. I just want to make expectations clear now on how we try to operate in situations like this, going forward 😄 Its not fun for anyone to deal with disagreements, but we gotta move forward somehow so we can keep working on the project we like FeelsOkayMan

Reply:
aye <img src="https://cdn.discordapp.com/emojis/982516941375823942.webp?size=56&quality=lossless" alt="OkaygeBusiness" height="24">

Sorry for forcing your hand on making these changes now. Still we're gonna have to figure it out soon!